share/smack/lib/smack.c

```diff
@@ -1090,61 +1197,80 @@ void __SMACK_decls(void) {
   D("function {:inline} $load.bvdouble(M: [ref] bvdouble, p: ref) returns (bvdouble) { M[p] }");
   D("function {:inline} $store.bvdouble(M: [ref] bvdouble, p: ref, v: bvdouble) returns ([ref] bvdouble) { M[p := v] }");
+  D("function {:inline} $load.bvlongdouble(M: [ref] bvlongdouble, p: ref) returns (bvlongdouble) { M[p] }");
+  D("function {:inline} $store.bvlongdouble(M: [ref] bvlongdouble, p: ref, v: bvlongdouble) returns ([ref] bvlongdouble) { M[p := v] }");
 
   D("function {:inline} $store.bytes.bvhalf(M:[ref]bv8, p:ref, v:bvhalf) returns ([ref]bv8) {"
     "$store.bytes.bv16(M, p, $bitcast.bvhalf.bv16(v))}");
   D("function {:inline} $store.bytes.bvfloat(M:[ref]bv8, p:ref, v:bvfloat) returns ([ref]bv8) {"
     "$store.bytes.bv32(M, p, $bitcast.bvfloat.bv32(v))}");
   D("function {:inline} $store.bytes.bvdouble(M:[ref]bv8, p:ref, v:bvdouble) returns ([ref]bv8) {"
     "$store.bytes.bv64(M, p, $bitcast.bvdouble.bv64(v))}");
+  D("function {:inline} $store.bytes.bvlongdouble(M:[ref]bv8, p:ref, v:bvlongdouble) returns ([ref]bv8) {"
+    "$store.bytes.bv80(M, p, $bitcast.bvlongdouble.bv80(v))}");
   D("function {:inline} $store.unsafe.bvhalf(M:[ref]i8, p:ref, v:bvhalf) returns ([ref]i8) {"
     "$store.i16(M, p, $bitcast.bvhalf.i16(v))}");
   D("function {:inline} $store.unsafe.bvfloat(M:[ref]i8, p:ref, v:bvfloat) returns ([ref]i8) {"
     "$store.i32(M, p, $bitcast.bvfloat.i32(v))}");
   D("function {:inline} $store.unsafe.bvdouble(M:[ref]i8, p:ref, v:bvdouble) returns ([ref]i8) {"
     "$store.i64(M, p, $bitcast.bvdouble.i64(v))}");
+  D("function {:inline} $store.unsafe.bvlongdouble(M:[ref]i8, p:ref, v:bvlongdouble) returns ([ref]i8) {"
+    "$store.i80(M, p, $bitcast.bvlongdouble.i80(v))}");
 
   D("function {:inline} $load.bytes.bvhalf(M: [ref] bv8, p: ref) returns (bvhalf) {"
     "$bitcast.bv16.bvhalf($load.bytes.bv16(M, p))}");
   D("function {:inline} $load.bytes.bvfloat(M: [ref] bv8, p: ref) returns (bvfloat) {"
     "$bitcast.bv32.bvfloat($load.bytes.bv32(M, p))}");
   D("function {:inline} $load.bytes.bvdouble(M: [ref] bv8, p: ref) returns (bvdouble) {"
     "$bitcast.bv64.bvdouble($load.bytes.bv64(M, p))}");
+  D("function {:inline} $load.bytes.bvlongdouble(M: [ref] bv8, p: ref) returns (bvlongdouble) {"
+    "$bitcast.bv80.bvlongdouble($load.bytes.bv80(M, p))}");
   D("function {:inline} $load.unsafe.bvhalf(M: [ref] i8, p: ref) returns (bvhalf) {"
     "$bitcast.i16.bvhalf($load.i16(M, p))}");
   D("function {:inline} $load.unsafe.bvfloat(M: [ref] i8, p: ref) returns (bvfloat) {"
     "$bitcast.i32.bvfloat($load.i32(M, p))}");
   D("function {:inline} $load.unsafe.bvdouble(M: [ref] i8, p: ref) returns (bvdouble) {"
     "$bitcast.i64.bvdouble($load.i64(M, p))}");
+  D("function {:inline} $load.unsafe.bvlongdouble(M: [ref] i8, p: ref) returns (bvlongdouble) {"
+    "$bitcast.i80.bvlongdouble($load.i80(M, p))}");
+  #endif
```

Comment:
Why did you add this `endif` here? Seems extraneous.

Reply:
There were merge conflicts, and I accidentally kept this line. I've now removed it.